src/server/game/AI/SmartScripts/SmartAI.cpp

```diff
@@ -889,7 +889,11 @@ void SmartAI::SetFollow(Unit* target, float dist, float angle, uint32 credit, ui
     _followArrivedEntry = end;
     _followCreditType = creditType;
     SetRun(_run);
-    me->GetMotionMaster()->MoveFollow(target, _followDist, _followAngle);
+
+    if (dist > 0.f && angle > 0.f)
```

Comment:
Not
```cpp
    if (dist > 0.f || angle > 0.f)
```
?

Reply:
No. Because the default follow behaivior only should kick in if no custom parameter is given

Reply:
It does not break the logic, but allows non-default movement (from SAI) if script uses only dist or only angle. Not all scripts uses (dist > 0 and angle > 0)